nautobot/core/views/mixins.py

```diff
@@ -249,7 +255,22 @@ def get_permissions_for_model(self, model, actions):
         """
         model_permissions = []
         for action in actions:
-            model_permissions.append(f"{model._meta.app_label}.{action}_{model._meta.model_name}")
+            # Append the base action permission if and only if it is valid and the model provided matches the queryset model.
+            if self.custom_view_base_action:
+                if self.custom_view_base_action not in BASE_ACTIONS:
+                    self.logger.error(
+                        f"Invalid base action '{self.custom_view_base_action}' specified in {self.__class__.__name__}. "
+                    )
+                elif model == self.get_queryset().model:
+                    model_permissions.append(
+                        f"{model._meta.app_label}.{self.custom_view_base_action}_{model._meta.model_name}"
+                    )
+
+            # Append additional object permissions if specified.
+            if self.custom_view_additional_permissions:
+                model_permissions.append(*self.custom_view_additional_permissions)
+            else:
+                model_permissions.append(f"{model._meta.app_label}.{action}_{model._meta.model_name}")
```

Comment:
Why is this an `else` case? It's not obvious/intuitive to me that `custom_view_ADDITIONAL_permissions` should override the default action permission.

Reply:
Thanks for catching this. You are right, `else` should not be there